modules/prbot/src/test/scala/ru/bugzmanov/prcheck/checks/PmdExecutorSpec.scala

```diff
@@ -0,0 +1,69 @@
+package ru.bugzmanov.prcheck.checks
+
+import java.io.{ByteArrayInputStream, InputStream}
+import java.nio.charset.Charset
+
+import net.sourceforge.pmd.util.datasource.DataSource
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+
+class PmdExecutorSpec extends FlatSpec {
+  import StringDataSource.fromString
+  val HelloWorldIssues = Seq("UseUtilityClass", "UseVarargs", "SystemPrintln", "NoPackage")
+  val executor = JavaPmdExecutor.fromRulesFile("pmd_rules.xml")
+
+  val BadHelloWorldCode = fromString(
+    name = "bad-hello-word",
+    contents = """
+    |public class HelloWorld {
+    |
+    |    public static void main(String[] args) {
+    |        System.out.println("Hello, World");
+    |    }
+    |}
+    """.stripMargin
+  )
+
+  val GoodHelloWorldCode = fromString(
+    name = "good-hello-world",
+    contents =
+      """
+      |package foo.bar;
+      |
+      |public final class HelloWorld {
+      |
+      |    private HelloWorld() { /* so PMD will think it's utility class */ }
+      |
+      |    public static void main(String ... args) {
+      |        log.info("Hello, World");
+      |    }
+      |}
+      """.stripMargin
+  )
+
+
+  "Pmd suite" should "report issues for problematic version of 'Hello world'" in {
+    val issues = executor.analyze(BadHelloWorldCode).get
+    issues.map(_.rule) should contain theSameElementsAs HelloWorldIssues
+  }
+  it should "remain silent for linted version of 'Hello world'" in {
+    val issues = executor.analyze(GoodHelloWorldCode).get
+    issues shouldBe empty
+  }
+  it should "correctly process multiple files at once" in {
```

Comment:
it would be nice to rephrase it

the idea was that executor should not accumulate state in between different processed files